ENIGMAsystem/SHELL/Platforms/General/POSIX/main.cpp

```diff
@@ -40,6 +42,25 @@ void initialize_directory_globals() {
 
 }
 
+namespace enigma_user {
+
+string shellscript_evaluate(string command, size_t buffer_size) {
+  char buffer[buffer_size];
+  FILE *file = popen(command.c_str(), "r");
+  while (fgets(buffer, buffer_size, file) != NULL) {}
```

Comment:
Have you tested this code? It shouldn't work for multiple lines of text, anyway.

Reply:
Well, it technically works for multiline output. Whether it does for the command argument I haven't tested, but that could be solved by doing a std::replace() all the new line characters (\n and \r individually) with a whitespace. My code I have tested with has no new lines but it does have stuff that would be easier to read if it had multiple lines such as if statements.

Reply:
It will not work for multi-line output, as implemented. You always write to the same buffer, right at the beginning. Since `fgets` null-terminates, it will "work" in the sense that the return value will always be the last line printed. Unless the command prints two newlines at the end instead of one, in which case, your code will return the empty string.

Reply:
well i can verify from my testing that the multi-line output 100% works, and i have seen quite a few new lines returned

Reply:
for instance, here's what selecting multiple files from an zenity open file dialog returns, and surprise-surprise it actually output the new lines:

![Screenshot from 2019-03-10 12-01-12](https://user-images.githubusercontent.com/4379204/54087726-4cf11200-432c-11e9-9e36-2952755ebb2e.png)

Reply:
The default separator in Zenity is the space character. It would be an operating system bug if your code worked the way you had written it.

Reply:
I'm not using the default separator. I changed it to \r

Reply:
(Which means there are no newlines in the output.)

Reply:
Oh no, but there are. You didn't see the shell script I was executing it appears, even though I sent you it like 5 times on hangouts

Reply:
I also know this is not an operating system bug because the TFD source uses fgets too many people have downloaded that thing and no one has complained about multiline output not working once

Reply:
Sam, you might as well argue with me that 2 + 2 is 17. It's impossible for your original implementation to read in a newline character. Each newline encountered, `fgets` will stop. You will call `fgets` again, and it will overwrite the previous line in the buffer. That is the end of the story.

Reply:
Ok fine, you win big daddy. But I still say 2 + 2 = 17